script/run-comp-match

```diff
@@ -0,0 +1,200 @@
+    controller_utils.get_mode_file().write_text('comp\n')
+    controller_utils.record_match_data(match_data)
+
+    for zone_id, tla in enumerate(match_data.teams):
+        zone_path = get_zone_path(zone_id)
+
+        if zone_path.exists():
+            shutil.rmtree(zone_path)
+
+        if tla is None:
+            # no team in this zone
+            continue
+
+        zone_path.mkdir()
+        with ZipFile(f'{archives_dir / tla}.zip') as zipfile:
+            zipfile.extractall(zone_path)
+
+
+def run_match() -> None:
+    try:
+        subprocess.check_call([
+            'webots',
+            '--batch',
+            '--stdout',
+            '--stderr',
+            '--minimize',
+            str(REPO_ROOT / 'worlds' / 'Arena.wbt'),
+        ])
+    except subprocess.CalledProcessError as e:
+        BOLD = '\033[1m'
+        FAIL = '\033[91m'
+        ENDC = '\033[0m'
+
+        log_text = controller_utils.get_competition_supervisor_log_filepath().read_text()
+
+        print("\n" + FAIL + log_text + ENDC)  # noqa:T001
+        print(  # noqa:T001
+            f"{BOLD}{FAIL}Simulation errored (exit code {e.returncode}). "
+            f"Competition supervisor logs are above.{ENDC}",
+        )
+        exit(1)
+
+
+def collate_logs(archives_dir: Path, match_data: controller_utils.MatchData) -> None:
+    """
+    Copy the teams' logs into directories next to their original code archives.
+    """
+
+    for zone_id, tla in enumerate(match_data.teams):
+        if tla is None:
+            # no team in this zone
+            continue
+
+        log_filname = controller_utils.get_robot_log_filename(zone_id)
+        log_path = get_zone_path(zone_id) / log_filname
+
+        team_dir = archives_dir / tla
+        team_dir.mkdir(exist_ok=True)
+
+        shutil.copy(log_path, team_dir)
+
+
+def archive_match_file(archives_dir: Path, match_data: controller_utils.MatchData) -> None:
+    """
+    Copy the match file (which contains the scoring data) to the archives directory.
+
+    This also renames the file to the name which SRComp expects.
+    """
+
+    matches_dir = archives_dir / 'matches'
+    matches_dir.mkdir(exist_ok=True)
+
+    # The file contains JSON data, we're relying here on the fact that JSON is a
+    # strict subset of YAML.
+    completed_match_file = matches_dir / f'{match_data.match_number:0>3}.yaml'
+
+    shutil.copy(controller_utils.get_match_file(), completed_match_file)
+
+
+def archive_match_recordings(archives_dir: Path) -> None:
+    """
+    Copy the recordings to the archives directory.
+    """
+
+    recordings_dir = archives_dir / 'recordings'
+    recordings_dir.mkdir(exist_ok=True)
+
+    recording_stem = controller_utils.get_recording_stem()
+
+    for path in recording_stem.parent.glob(f'{recording_stem.name}.*'):
+        shutil.copy(path, recordings_dir)
+
+    try:
+        shutil.copytree(recording_stem.parent / 'textures', recordings_dir / 'textures')
```

Comment:
Worth a comment here about why we're copying textures?

Reply:
2a83607